pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -881,8 +908,9 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
+// getComputeIDbyHostname tries to find ID of the OpenStack instance where
+// this code runs.
 func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
```

Comment:
Ah... So I _think_ the code below no longer works with your `kubernetes-master-3-6050` example since I don't see anything that ignores an arbitrary suffix.  In general, the hostname might have nothing at all to do with the openstack instance name, so if the goal is just to find the instance ID of the current host, then it would be good to explore other approaches entirely.

Perhaps we should just read `/var/lib/cloud/data/instance-id` and/or do the same config-drive or metadata lookup  ourselves?

The code elsewhere in this current change looks fine from a code pov, but I'm beginning to feel the original motivation for it is perhaps not the best direction to pursue.  Would reading instance-id from config-drive / metadata server address the original need?